File: src/main/java/mods/cybercat/gigeresque/client/entity/render/AquaticChestbursterEntityRenderer.java
package mods.cybercat.gigeresque.client.entity.render;

import com.mojang.blaze3d.vertex.PoseStack;

import mods.cybercat.gigeresque.client.entity.model.AquaticChestbursterEntityModel;
import mods.cybercat.gigeresque.client.entity.render.feature.AquaBusterBloodFeatureRenderer;
import mods.cybercat.gigeresque.common.entity.impl.AquaticChestbursterEntity;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import software.bernie.geckolib.renderer.GeoEntityRenderer;

@Environment(EnvType.CLIENT)
public class AquaticChestbursterEntityRenderer extends GeoEntityRenderer<AquaticChestbursterEntity> {
	public AquaticChestbursterEntityRenderer(EntityRendererProvider.Context context) {
		super(context, new AquaticChestbursterEntityModel());
		this.shadowRadius = 0.5f;
		this.addRenderLayer(new AquaBusterBloodFeatureRenderer(this));
	}

	@Override
	public void render(AquaticChestbursterEntity entity, float entityYaw, float partialTicks, PoseStack stack,
			MultiBufferSource bufferIn, int packedLightIn) {
		float scaleFactor = 1.0f + (entity.getGrowth() / entity.getMaxGrowth());
		stack.scale(scaleFactor, scaleFactor, scaleFactor);
		super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn);
	}

	@Override
	protected float getDeathMaxRotation(AquaticChestbursterEntity entityLivingBaseIn) {
		return 0;
	}
}

File: src/main/kotlin/com/bvanseg/gigeresque/client/entity/render/AquaticChestbursterEntityRenderer.kt
//package com.bvanseg.gigeresque.client.entity.render
//
//import com.bvanseg.gigeresque.client.entity.model.AquaticChestbursterEntityModel
//import com.bvanseg.gigeresque.client.extensions.scale
//import com.bvanseg.gigeresque.common.entity.impl.AquaticChestbursterEntity
//import net.fabricmc.api.EnvType
//import net.fabricmc.api.Environment
//import net.minecraft.client.render.VertexConsumerProvider
//import net.minecraft.client.render.entity.EntityRendererFactory
//import net.minecraft.client.util.math.MatrixStack
//import software.bernie.geckolib3.renderers.geo.GeoEntityRenderer
//
///**
// * @author Boston Vanseghi
// */
//@Environment(EnvType.CLIENT)
//class AquaticChestbursterEntityRenderer(context: EntityRendererFactory.Context) :
//    GeoEntityRenderer<AquaticChestbursterEntity>(
//        context,
//        AquaticChestbursterEntityModel()
//    ) {
//    init {
//        this.shadowRadius = 0.1f
//    }
//
//    override fun render(
//        entity: AquaticChestbursterEntity,
//        entityYaw: Float,
//        partialTicks: Float,
//        stack: MatrixStack,
//        bufferIn: VertexConsumerProvider,
//        packedLightIn: Int
//    ) {
//        stack.scale(1.0f + (entity.growth / entity.maxGrowth))
//        super.render(entity, entityYaw, partialTicks, stack, bufferIn, packedLightIn)
//    }
//}